deployment/ccip/changeset/deployer_group.go

```diff
@@ -30,11 +34,59 @@ type DeployerGroup struct {
 	describeContext   *proposalutils.ArgumentContext
 }
 
-type DescribedTransaction struct {
+type DescribedTransaction interface {
+	Describe() string
+	ToMCMS(selector uint64) (mcmstypes.Transaction, error)
+}
+
+type EvmDescribedTransaction struct {
 	Tx          *types.Transaction
 	Description string
 }
 
+func (d EvmDescribedTransaction) Describe() string {
+	return d.Description
+}
+
+func (d EvmDescribedTransaction) ToMCMS(selector uint64) (mcmstypes.Transaction, error) {
+	return proposalutils.TransactionForChain(selector, d.Tx.To().Hex(), d.Tx.Data(), d.Tx.Value(), "", []string{})
+}
+
+type SolanaDescribedTransaction struct {
+	Tx           solana.Instruction
+	ProgramID    string
+	ContractType deployment.ContractType
+	Description  string
+}
+
+func (d SolanaDescribedTransaction) Describe() string {
+	return d.Description
+}
+
+func (d SolanaDescribedTransaction) ToMCMS(selector uint64) (mcmstypes.Transaction, error) {
+	data, err := d.Tx.Data()
+	if err != nil {
+		return mcmstypes.Transaction{}, fmt.Errorf("failed to extract data: %w", err)
+	}
+	for _, account := range d.Tx.Accounts() {
+		if account.IsSigner {
+			account.IsSigner = false
```

Comment:
Why is this being done? Maybe just a comment

Reply:
Good point added a comment on the line. This is something I picked from another Solana MCMS utilities. It's because the signer is a different PDA when executed through Solana